components/you_need_a_budget/actions/update-transaction/update-transaction.mjs

```diff
@@ -0,0 +1,105 @@
+import app from "../../you_need_a_budget.app.mjs";
+
+export default {
+  key: "you_need_a_budget-update-transaction",
+  name: "Update Transaction",
+  description: "Updates existing transaction. [See the docs](https://api.youneedabudget.com/v1#/Transactions/updateTransaction)",
+  version: "0.0.5",
+  type: "action",
+  props: {
+    app,
+    budgetId: {
+      propDefinition: [
+        app,
+        "budgetId",
+      ],
+    },
+    accountId: {
+      propDefinition: [
+        app,
+        "accountId",
+        (c) => ({
+          budgetId: c.budgetId,
+        }),
+      ],
+    },
+    transactionId: {
+      type: "string",
+      label: "Transaction ID",
+      description: "The ID of the transaction to update",
+    },
+    date: {
+      propDefinition: [
+        app,
+        "date"
+      ],
+    },
+    payee: {
+      propDefinition: [
+        app,
+        "payee",
+        (c) => ({
+          budgetId: c.budgetId,
+        })
+      ],
+    },
+    categoryId: {
+      propDefinition: [
+        app,
+        "categoryId",
+        (c) => ({
+          budgetId: c.budgetId,
+          optional: true,
+        }),
+      ],
+    },
+    amount: {
+      propDefinition: [
+        app,
+        "amount"
+      ],
+    },
+    memo: {
+      type: "string",
+      label: "Memo",
+      description: "A short description",
+      optional: true,
+    },
+    cleared: {
+      type: "boolean",
+      label: "Cleared",
+      description: "Marking a transaction as cleared means it has posted to your bank account",
```

Comment:
```suggestion
      description: "Mark a transaction as cleared when it has posted to your bank account.",
```

Reply:
Changed "Marking" to "Mark" and rephrased the sentence to follow Google's Developer docs style guide (use present tense and active voice).